apps/settings/js/modules/base/task_scheduler.js

```
@@ -0,0 +1,139 @@
+/**
+ * TaskScheduler helps manage tasks and ensures they are executed in
+ * sequential order. When a task of a certain type is enqueued, all pending
+ * tasks of the same type in the queue are removed. This avoids redundant
+ * queries and improves user perceived performance.
+ *
+ * @module Task_sheduler
+ */
+define(function(require) {
+  'use strict';
+
+  var Defer = require('modules/defer');
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  
+  /**
+   * @class TaskScheduler
+   * @returns {TaskScheduler}
+   */
+  var TaskScheduler = Module.create(function() {
```

Maybe I should throw a exception in `Module` in the case. :p